### PR TITLE
Comparison

### PR DESCRIPTION
**Pull request by palpatel0**
comparing the data across the data sets for consistency
also trying to make descriptive stats for the offences per borough

### CODE REVIEW

File: comparison.Rmd [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+### Comparing the same offence by borough
+
+```{python}
+pd.DataFrame(data.borough.loc[:, 'Arson', :].T.groupby(level='year').sum().sum())
+```
+
+The above dataframe shows the total number of Arson offences across the entirety of the `borough` dataset.
+
+```{python}
+pd.DataFrame(((data.borough.loc[:, 'Arson', :].T.groupby(level='year').sum().sum()) - (data.ward.loc[:, :, :, 'Arson', :].groupby(level=[3]).sum().T.sum())).sort_values(ascending=False))
```
Comment (TomWFox) [inline comparison.Rmd:109]:
```suggestion
borough_arson = (data.borough
                     .loc[:, 'Arson', :]
                     .T
                     .groupby(level='year')
                     .sum()
                     .sum())
ward_arson = (data.ward
                  .loc[:, :, :, 'Arson', :]
                  .groupby(level=[3])
                  .sum()
                  .T
                  .sum())

pd.DataFrame((borough_arson - ward_arson).sort_values(ascending=False))
```

File: comparison.Rmd [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Heathrow Airport is in the Hillingdon borough and London City Airport is in Newham. This means that the Arson offences are being split between these two boroughs, leading to less reports in the `borough` dataset compared to the ward dataset for the Hillingdon and Newham boroughs.
+
+The `ward` dataset does include a `Aviation Security (SO18)` borough, however. This is in reference to the Heathrow Airport.
+
+```{python}
+borhil = (data.borough.loc[:, 'Arson', ['Newham', 'Hillingdon']].T.groupby(level='year').sum().T.sum()).to_frame('Arson')
+halca = (data.borough.loc[:, 'Arson', 'London Heathrow and London City Airports'].T.groupby(level='year').sum())
+wardhil = (data.ward.loc[:, :, :, 'Arson', ['Newham','Hillingdon','Aviation Security (SO18)']].sum().T.groupby(level='year').sum()).to_frame('Arson Offences')
+((borhil['Arson'] + halca['Arson and Criminal Damage']) - wardhil['Arson Offences']).sum()
```
Comment (TomWFox) [inline comparison.Rmd:124]:
```suggestion
bor_hillingdon = ((data.borough
                       .loc[:, 'Arson', ['Newham', 'Hillingdon']]
                       .T
                       .groupby(level='year')
                       .sum()
                       .T
                       .sum())
                       .to_frame('Arson'))
    
bor_heathrow_londoncity = (data.borough
                               .loc[:, 'Arson', 'London Heathrow and London City Airports']
                               .T
                               .groupby(level='year')
                               .sum())

ward_hillingdon = ((data.ward
                        .loc[:, :, :, 'Arson', ['Newham','Hillingdon','Aviation Security (SO18)']]
                        .sum()
                        .T
                        .groupby(level='year')
                        .sum())
                        .to_frame('Arson Offences'))

((bor_hillingdon['Arson'] + bor_heathrow_londoncity['Arson and Criminal Damage']) - ward_hillingdon['Arson Offences']).sum()
```

File: comparison.Rmd [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+This leaves a difference of `12.0`.
+
+This means that there is more reports in the `borough` dataset than the `ward` dataset.
+
+```{python}
+borhil1 = (data.borough.loc[:, 'Arson', 'Hillingdon'].T.groupby(level='year').sum().T.sum()).to_frame('Arson')
+halca1 = (data.borough.loc[:, 'Arson', 'London Heathrow and London City Airports'].T.groupby(level='year').sum())
+wardhil1 = (data.ward.loc[:, :, :, 'Arson', ['Hillingdon','Aviation Security (SO18)']].sum().T.groupby(level='year').sum()).to_frame('Arson Offences')
+((borhil1['Arson'] + halca1['Arson and Criminal Damage']) - wardhil1['Arson Offences']).sum()
```
Comment (TomWFox) [inline comparison.Rmd:146]:
```suggestion
bor_hillingdon1 = ((data.borough
                        .loc[:, 'Arson', 'Hillingdon']
                        .T
                        .groupby(level='year')
                        .sum()
                        .T
                        .sum())
                        .to_frame('Arson'))

ward_hillingdon1 = ((data.ward
                         .loc[:, :, :, 'Arson', ['Hillingdon','Aviation Security (SO18)']]
                         .sum()
                         .T
                         .groupby(level='year')
                         .sum())
                         .to_frame('Arson Offences'))

((bor_hillingdon1['Arson'] + bor_heathrow_londoncity['Arson and Criminal Damage']) - ward_hillingdon1['Arson Offences']).sum()
```

File: comparison.Rmd
Comment (TomWFox) [inline comparison.Rmd:1]: Can you go through and change where you've used `code_snippets` to __bold__ or _italic_?

Obviously you can use code snippet but only for code e.g. `data.ward`

**Review by TomWFox — CHANGES_REQUESTED** [review]
Just needs some tidying up. Otherwise looking good.

**Comment by TomWFox** [comment]
Reminder: update any hard-coded numbers based on addition of November data

**Comment by palpatel0** [comment]
Made all changes as requested. Ready for review :)

**Review by palpatel0 — COMMENTED** [review]
changes made

**Review by TomWFox — APPROVED** [review]
LGTM!

**Pull request merged**